src/cart/crt_group.c

```diff
@@ -3370,6 +3370,47 @@ crt_register_eviction_cb(crt_eviction_cb cb, void *arg)
 	return rc;
 }
 
+int
+crt_register_event_cb(crt_event_cb event_handler, void *arg)
+{
+	struct crt_event_cb_priv	*cb_priv;
+	int				 rc = 0;
+
+	D_ALLOC_PTR(cb_priv);
+	if (cb_priv == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
+	cb_priv->cecp_func = event_handler;
+	cb_priv->cecp_args = arg;
+
+	/* Try to unregister this handler first to avoid duplication */
+	crt_unregister_event_cb(event_handler, arg);
```

Comment:
i dont think we should unregister, we should check instead and return error to user. the purpose of this is to provide user with notification when they attempt to register twice, as it would indicate some error on user side. if we just unregister/re-register, we once again hide that quietly from the user